code/game/machinery/big_manipulator.dm

```diff
@@ -0,0 +1,261 @@
+/obj/machinery/big_manipulator
+	name = "Big Manipulator"
+	desc = "Take and drop objects. Innovation..."
+	icon = 'icons/obj/machines/big_manipulator_parts/big_manipulator_core.dmi'
+	icon_state = "core"
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/big_manipulator
+	greyscale_colors = "#d8ce13"
+	greyscale_config = /datum/greyscale_config/big_manipulator
+	/// How many time manipulator need to take and drop item.
+	var/working_speed = 2 SECONDS
+	/// Using high tier manipulators speeds up big manipulator and requires more energy.
+	var/power_use_lvl = 0.2
+	/// When manipulator already working with item inside he don't take any new items.
+	var/on_work = FALSE
+	/// Activate mechanism.
+	var/on = FALSE
+	/// Dir to get turf where we take items.
+	var/take_here = NORTH
+	/// Dir to get turf where we drop items.
+	var/drop_here = SOUTH
+	/// Turf where we take items.
+	var/turf/take_turf
+	/// Turf where we drop items.
+	var/turf/drop_turf
+	/// Obj inside manipulator.
+	var/datum/weakref/containment_obj
+	/// Other manipulator component.
+	var/obj/effect/manipulator_hand
+
+/obj/machinery/big_manipulator/Initialize(mapload)
+	. = ..()
+	take_turf = get_step(src, take_here)
+	drop_turf = get_step(src, drop_here)
+	create_manipulator_hand()
+	RegisterSignal(manipulator_hand, COMSIG_QDELETING, PROC_REF(on_hand_qdel))
+	manipulator_lvl()
+
+/obj/machinery/big_manipulator/Destroy(force)
+	. = ..()
+	qdel(manipulator_hand)
+	if(isnull(containment_obj))
+		return
+	var/obj/obj_resolve = containment_obj?.resolve()
+	obj_resolve.forceMove(get_turf(obj_resolve))
```

Comment:
Shouldn't you check if there's an obj_resolve before forceMoving it? Seems like it could be null, you even check if there's a containment object in the first place